tests/test_project_management.py

```diff
@@ -208,9 +309,476 @@ def test_create_android_app_polling_failure(self):
+
+        ios_apps = project_management.list_ios_apps()
+
+        expected_app_ids = set(['1:12345678:ios:ca5cade5', '1:12345678:ios:ca5cade5cafe'])
+        assert set(app.app_id for app in ios_apps) == expected_app_ids
+        assert len(recorder) == 1
+        self._assert_request_is_correct(recorder[0], 'GET', TestListIosApps._LISTING_URL)
+
+    def test_list_ios_apps_rpc_error(self):
+        recorder = self._instrument_service(statuses=[503], responses=['some error response'])
+
+        with pytest.raises(project_management.ApiCallError) as excinfo:
+            project_management.list_ios_apps()
+
+        assert 'Backend servers are over capacity' in str(excinfo.value)
+        assert excinfo.value.detail is not None
+        assert len(recorder) == 1
+
+    def test_list_ios_apps_multiple_pages(self):
+        recorder = self._instrument_service(
+            statuses=[200, 200],
+            responses=[LIST_IOS_APPS_PAGE_1_RESPONSE, LIST_IOS_APPS_PAGE_2_RESPONSE])
+
+        ios_apps = project_management.list_ios_apps()
+
+        expected_app_ids = set(['1:12345678:ios:ca5cade5', '1:12345678:ios:ca5cade5cafe'])
+        assert set(app.app_id for app in ios_apps) == expected_app_ids
+        assert len(recorder) == 2
+        self._assert_request_is_correct(recorder[0], 'GET', TestListIosApps._LISTING_URL)
+        self._assert_request_is_correct(recorder[1], 'GET', TestListIosApps._LISTING_PAGE_2_URL)
+
+    def test_list_ios_apps_multiple_pages_rpc_error(self):
+        recorder = self._instrument_service(
+            statuses=[200, 503],
+            responses=[LIST_IOS_APPS_PAGE_1_RESPONSE, 'some error response'])
+
+        with pytest.raises(project_management.ApiCallError) as excinfo:
+            project_management.list_ios_apps()
+
+        assert 'Backend servers are over capacity' in str(excinfo.value)
+        assert excinfo.value.detail is not None
+        assert len(recorder) == 2
+
+
+class TestAndroidApp(BaseProjectManagementTest):
+    _GET_METADATA_URL = ('https://firebase.googleapis.com/v1beta1/projects/-/androidApps/'
+                         '1:12345678:android:deadbeef')
+    _SET_DISPLAY_NAME_URL = ('https://firebase.googleapis.com/v1beta1/projects/-/androidApps/'
+                             '1:12345678:android:deadbeef?updateMask=displayName')
+    _GET_CONFIG_URL = ('https://firebase.googleapis.com/v1beta1/projects/-/androidApps/'
+                       '1:12345678:android:deadbeef/config')
+    _ADD_CERT_URL = ('https://firebase.googleapis.com/v1beta1/projects/-/androidApps/'
+                     '1:12345678:android:deadbeef/sha')
+    _LIST_CERTS_URL = ('https://firebase.googleapis.com/v1beta1/projects/-/androidApps/'
+                       '1:12345678:android:deadbeef/sha')
+    _DELETE_SHA_1_CERT_URL = 'https://firebase.googleapis.com/v1beta1/{0}'.format(SHA_1_NAME)
+    _DELETE_SHA_256_CERT_URL = 'https://firebase.googleapis.com/v1beta1/{0}'.format(SHA_256_NAME)
+
+    @pytest.fixture
```

Comment:
Nit: Can this be a class-level constant instead of a fixture?

Reply:
Yeah I thought about that too, but then we have to write `TestAndroidApp.ANDROID_APP` instead of just `android_app`. We can make this a module-level constant, but you have to scroll up farther as you said in the previous PR. I actually attempted to make many more things a class-level `@fixture` for this reason, until I realized even that's not free: the parameter list for each test grows. So I chose to use `@fixture` with the only item that every single test in this class uses: the app.